packages/core/src/request-builder/header-builder/headers-util.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { MapType } from '@sap-cloud-sdk/util';
+
+export function isNullish(x: any): x is null | undefined {
+  return x === null || x === undefined;
+}
+
+export function toHeaderObject(key: string, value: any): MapType<any> {
+  return isNullish(value) ? {} : { [key]: value };
+}
+
+export function getHeader(
```

Comment:
This does actually more then the name suggests, case is ignored and a {} is returned in case it is not the an empty object is given. Perhaps you can reflect in the name or check if the ignoring case etc. is really needed.

Reply:
I don't quite understand. In case a matching header is found an object containing this header is returned (using its original name). In case it is not, an empty object is returned.

Reply:
Ok, I see, you don't mean `getHeader' but `toHeaderObject`. I partially agree, I just took it over, this was there before. What do you think of `toSanitizedHeaderObject`?

Reply:
No I meant to getHeader() if I call it for getHeader('KEY_IN_CAPITAL',header) I can get as a return {'key_in_capital':'someValue;} which is sort of surprising. So you can also add some JS doc in the util just telling the user that case is ignored and {} is returned in case no key is found in contrast to an exception beeing thrown.

Reply:
Added documentation